docs/Relayer Fee Market.md

```diff
@@ -0,0 +1,92 @@
+# [A MVP Design of a Relayer Fee Market]
+
+## Requirements
+
+1. Users use native tokens of the source chain as the only method of payment;
+2. The estimated fee of the message to be transferred is supported by a pricing system on the source chain which is connected to a Relayer Fee Market;
+3. The cost on the target chain is at the expense of the relayer who claims the handling fee on the source chain with the proof of delivery after delivery is completed successfully. To incentivize relayers, the pricing system should ensure the gain is greater than the cost. If an automatic pricing mechanism is infeasible, relayers should be able to give their offers manually and shoulder the cost of offering. The relayers should be pushed if they fail to relay the message as expected.
+4. Relayers and Users constitute a secondary supply-and-demand market, where prices rise when supply is low and fall when supply is abundant. There are no access restrictions for relayers, and anyone can enter. Relayer should evaluate and quote at their own discretion as an economically rational person. An incomplete list of risks that relayers should take into account is as follows:
+    1. Fluctuation in token prices and exchange ratios;
+    2. Time delay between quoting and claiming;
+    3. Loss of staking funds due to software or network failures
+
+## Proposal A—Three-tiered Quotation
+
+This approach is suitable for scenarios with lower gas fee on the source chain and shorter finality time. It has better versatility, reliability, and robustness. Such networks include Heco, BSC, Polygon and Darwinia.
+
+First, the relayer posts their quotes based on the reference price and the expected profit on the blockchain at any time. An off-chain pricing system maintains the reference price. Each relayer should lock a sufficient default margin on the chain to guarantee the faithful execution of the deal.
+
+In this way, a series of ***Ask*** prices come into being in the ascending order on the blockchain. When the user initiates a request on the source chain, the three lowest offers **_P1<P2<P3_** are filtered out and $P3$  is used as the billing price.  The user can request a cross-chain message delivery after paying **_P3_**.
+
+Then the system requires the three relayers with respect to **_P1_**, **_P2_**, and **_P3_** to execute the delivery during three consecutive time slots: **_0~T1_**,  **_T1~T2_**, and **_T2~T3_** respectively, where **_T1_**, **_T2_**, and **_T3_** can be configured suitable for the scenario. If the relayer fails to deliver in their allotted time slot, the task goes to the next relayer. The penalty fee **_S(t)_** from the previous relayer also adds up to the gain of the current relayer as a bonus, where **_S(t)_** is a function of the delay **_t_**.
```

Comment:
After reminder of @AsceticBear , this is not the requirement of protocol during message send and delivery process.

It is only the incentive goals that will result from the delivery confirmation and relayer reward strategy.

Need some revise further to make this(multi candidate with priority design) more clear:

1. Redundancy of delivery order candidtes.
2. Priority result from ask prices
3. Coodination of relayers, by define approximate stratey for relayer reward in different time slot.

Reply:
I will update this doc when the fee market reward strategy concrete design is complete later this week.